frontend/src/components/CommentVotes.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { fetchUpvotesForComment, fetchDownvotesForComment } from '../store/actions/comments';
+import {Dispatch} from "redux";
+
+interface StateProps {
+    commentUpvotes: any,
+    commentDownvotes: any,
+}
+
+interface DispatchProps {
+    fetchUpvotesForComment: (commentId: number) => void,
+    fetchDownvotesForComment: (commentId: number) => void,
+}
+
+interface OwnProps {
+    commentId: number,
+    submitVote: (comment: {commentId: number, isUpvote: boolean}) => void,
+}
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+enum VoteType {
+    Upvote = 'commentUpvotes',
+    Downvote = 'commentDownvotes',
+}
+
+class CommentVotesComponents extends React.Component<Props> {
+
+    componentWillMount() {
+        this.props.fetchUpvotesForComment(this.props.commentId);
+        this.props.fetchDownvotesForComment(this.props.commentId);
+        setInterval(_ => {
+            this.props.fetchUpvotesForComment(this.props.commentId);
+            this.props.fetchDownvotesForComment(this.props.commentId);
+        }, 5000);
+    }
+
+    getCommentVotes = (voteType: VoteType) => {
+        const propVotes = this.props[voteType][this.props.commentId];
+        if (propVotes === undefined || propVotes === null) {
+            return [];
+        }
+        return propVotes;
+    };
+
+    submitVote = (isUpvote: boolean) => {
+        this.props.submitVote({
+            commentId: this.props.commentId,
+            isUpvote
+        })
+    };
+
+    render() {
+        const upvotes = this.getCommentVotes(VoteType.Upvote);
+        const downvotes = this.getCommentVotes(VoteType.Downvote);
+        return (
+            <div>
+                <div>{upvotes.length} upvotes</div>
+                <div>{downvotes.length} downvotes</div>
+                <button onClick={() => {this.submitVote(true)}}>Upvote</button>
+                <button onClick={() => {this.submitVote(false)}}>Downvote</button>
+            </div>
+        );
+    }
+}
+
+const mapStateToProps = (state: any): StateProps => ({
```

Comment:
this should not be any I think, rule of thumb -> you should have a really good reason to use any, and put a comment there

Reply:
I probably forgot to put `State` here.

Btw, the `State` has something like `web3Context: Web3Context | null` and what annoys me is that the component always needs to handle the case when `web3Context` is `null`. Is there any way to either make all the fields in redux non-null (`web3Context` is initialized by a call to the API so in the initial state it's `null`); or to somehow block the component from instantiating when `web3Context` is `null` so I don't have to include all the checks?